### DISCUSSION TITLE
[Feature] Improving performance when uploading large library

### DISCUSSION THREAD

**Discussion by Pataclop** [immich-app/immich#5100] [Feature Request]
When importing large number of images (20k) using chrome , perfomance is abysmal  when the "uploader GUI" is expanded (more than 30s per image). I even get the "chrome stopped working" message. 
But when the upload GUI is reduced to the cloud icon with the image count, it ll goes at normal speed (several images per second). I'm guessing it's trying to render the full 20k items and their individual progress bars, even if they are not in view. Maybe a fix could be to just render/display the active and failed jobs, and not the full list? 

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
The web doesn't mean to upload more than a handful of dozen images. Please use the CLI tool for bulk upload
